spec/stable-release.md

```diff
@@ -0,0 +1,62 @@
+# Stable Release
+Implement a "Release" mechanism for installing "consistent" versions of 
+Crucible.
+
+## Problem description
+We need to be able to install a "version" of Crucible that is "locked down" 
+and won't change at all if it is installed over time.
+
+Crucible installations cannot be used for regression testing in a consistent
+manner.
+
+bin/update will override the installed repository by pulling the latest commit
+if the installed commit is an older one.
+
+Crucible installer clones Crucible repository and all sub-projects from the
+latest "main" branch. A default installation contains all the Crucible 
+repositories from the latest merged commits.
+
+```
+#project-name        project-type   git-repo-url                     branch
+rickshaw             core           /rickshaw                        master
+multiplex            core           /multiplex                       master
+roadblock            core           /roadblock                       master
+...
+hwlatdetect          benchmark      /bench-hwlatdetect               main
+tracer               benchmark      /bench-tracer                    main
+...
+ftrace               tool           /tool-ftrace                     master
+testing              doc            /testing-repo                    master
```

Comment:
Do you think we should switch to the "new" subproject config idea (JSON) as part of this?

Reply:
I think so. Not a dependency, just a convenience, as we are changing crucible config/default_projects.